### PR TITLE
stop ClusterClient ResponseTunnel after first reply when ask is used, #22093

### PR DESCRIPTION
**Pull request by patriknw**
Refs #22093

### CODE REVIEW

**Comment by patriknw** [comment]
created issue for unrelated failure: https://github.com/akka/akka/issues/22097

**Comment by patriknw** [comment]
PLS BUILD

File: akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala [hunk 1]
```diff
@@ -812,12 +812,18 @@ object ClusterReceptionist {
      */
     class ClientResponseTunnel(client: ActorRef, timeout: FiniteDuration) extends Actor with ActorLogging {
       context.setReceiveTimeout(timeout)
+
+      private val isAsk = client.path.elements.headOption.contains("temp")
```
Comment (ktoso) [inline akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala]: maybe `client.path.toString startsWith "/temp/$` would be a bit future-safer?
I was checking if we don't register FunctionRefs here but seems we don't... Do you think this is something to worry about or not really? Not sure if Typed needs to register temp actors hm

**Review by ktoso — APPROVED** [review]
LGTM, somewhat iffy about the temp checking though I see that it currently works.

File: akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala (hunk 1 above)
Comment (patriknw) [inline akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala]: ok, we can add the additional check that the second element starts with "$"
(I prefer to not do full toString)

Comment (patriknw) [inline akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala]: done

File: akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala [hunk 2]
```diff
@@ -812,12 +812,21 @@ object ClusterReceptionist {
      */
     class ClientResponseTunnel(client: ActorRef, timeout: FiniteDuration) extends Actor with ActorLogging {
       context.setReceiveTimeout(timeout)
+
+      private val isAsk = {
+        val pathElements = client.path.elements
+        pathElements.size == 2 && pathElements.head == "temp" && pathElements.tail.head.startsWith("$")
+      }
```
Comment (ktoso) [inline akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala]: Cool, this seems more future ready :+1:

**Review by ktoso — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
PLS BUILD

**Pull request merged**